File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> dict;
        for(int i=0;i<nums.size();i++) {
            if(dict.count(nums[i]) > 0) {
                return true;
            }
            dict[nums[i]] = i;
        }

        return false;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = nums[0]

        max_ending = nums[0]

        for i in range(1, len(nums)):

            max_ending = max(nums[i], max_ending + nums[i])

            res = max(res, max_ending)
        
        return res